src/main/java/use_cases/DatabaseManager.java

```diff
@@ -50,313 +26,21 @@ public void addNewPost(Post newPost) {
      * added (there was no user with the same username). False if
      * unsuccessful.
      */
-    public boolean addNewUser(User newUser) {
-        try {
-            String query = "INSERT INTO `user_info`(`user_id`, `username`, `password`, `bio`) " +
-                    "VALUES (?,?,?,?)";
-
-            PreparedStatement preparedStmt = connection.prepareStatement(query);
-            preparedStmt.setString(1, newUser.getId());
-            preparedStmt.setString(2, newUser.getUsername());
-            preparedStmt.setString(3, newUser.getPassword());
-            preparedStmt.setString(4, newUser.getBio());
-
-            preparedStmt.execute();
-            return true;
-        } catch (Exception e) {
-            e.printStackTrace();
-            return false;
-        }
-    }
+    public abstract boolean addNewUser(User newUser);
 
     /**
      * Gets all the posts stored in the database.
      * @return an Array of all the posts stored in the database.
      */
-    public Post[] getAllPosts() {
-        try {
-            HashMap<String, Post> postData = new HashMap<>();
-            HashMap<String, String> postIdsToRecipeIds = new HashMap<>();
-            String postsQuery = "SELECT * FROM `posts` WHERE 1";
-            ResultSet postsResult = this.connection.createStatement().executeQuery(postsQuery);
-            while(postsResult.next()) {
-                String postId = postsResult.getString("post_id");
-                String userId = postsResult.getString("user_id");
-                String recipeId = postsResult.getString("recipe_id");
-                String category = postsResult.getString("category");
-                LocalDateTime postedTime = postsResult.getTimestamp("posted_time").toLocalDateTime();
-                postData.put(
-                        postId,
-                        new Post(userId,
-                            postedTime,
-                            new Recipe("", new ArrayList<>(), new ArrayList<>()),
-                            category)
-                    );
-            }
+    public abstract Post[] getAllPosts();
 
-            // TODO: add recipe data
+    public abstract boolean updatePost(Post updatedPost);
```

Comment:
Here I believe it should be a { at the end not a ; because it's a method header

Reply:
I think that's the correct syntax for abstract methods, like the example here https://www.w3schools.com/java/java_abstract.asp
Please let me know if I've made a mistake

Reply:
oh yeah you are right my bad. I think everything looks good then! :)